bundles/org.openhab.binding.pixometer/src/main/java/org/openhab/binding/pixometer/handler/MeterHandler.java

```diff
@@ -0,0 +1,308 @@
+import org.openhab.binding.pixometer.internal.config.ReadingInstance;
+import org.openhab.binding.pixometer.internal.data.MeterState;
+import org.openhab.binding.pixometer.internal.serializer.CustomReadingInstanceDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link MeterHandler} is responsible for handling data and measurements of a meter thing
+ *
+ * @author Jerome Luckenbach - Initial contribution
+ */
+@NonNullByDefault
+public class MeterHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MeterHandler.class);
+
+    private static final String API_VERSION = "v1";
+    private static final String API_METER_ENDPOINT = "meters";
+    private static final String API_READINGS_ENDPOINT = "readings";
+
+    private final GsonBuilder gsonBuilder = new GsonBuilder().registerTypeAdapter(ReadingInstance.class,
+            new CustomReadingInstanceDeserializer());
+    private final Gson gson = gsonBuilder.create();
+    private final JsonParser jsonParser = new JsonParser();
+
+    private @NonNullByDefault({}) String resourceID;
+    private @NonNullByDefault({}) String meterID;
+    private @NonNullByDefault({}) ExpiringCache<@Nullable MeterState> cache;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    public MeterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof RefreshType) {
+                updateMeter(channelUID, cache.getValue());
+            } else {
+                logger.debug("The pixometer binding is read-only and can not handle command {}", command);
+            }
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Pixometer handler '{}'", getThing().getUID());
+        updateStatus(ThingStatus.UNKNOWN);
+
+        PixometerMeterConfiguration config = getConfigAs(PixometerMeterConfiguration.class);
+        setRessourceID(config.resourceId);
+
+        cache = new ExpiringCache<@Nullable MeterState>(Duration.ofMinutes(60), this::refreshCache);
+
+        Bridge b = this.getBridge();
+        if (b == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Could not find bridge (pixometer config). Did you choose one?");
+            return;
+        }
+
+        obtainMeterId();
+
+        // Start polling job with the interval, that has been set up in the bridge
+        int pollingPeriod = Integer.parseInt(b.getConfiguration().get(CONFIG_BRIDGE_REFRESH).toString());
```

Comment:
I am not fully sure, if the framework validates the value or if you should better check it here to be >=60.
By the way, one hours seems incredibly long for a minimal refresh rate. Is there a request limit on the API that makes this necessary?

Reply:
>By the way, one hours seems incredibly long for a minimal refresh rate. Is there a request limit on the API that makes this necessary?

Nope, i have seen simply no need in requesting their API unnecessary.
Those values are based on my usage of the pixometer app and i try to get one meter reading per month per meter.
With this background 60 minutes seemed "fast enough" for me.

>I am not fully sure, if the framework validates the value or if you should better check it here to be >=60.

I am pretty sure that i had to manually edit the thing xml to set a lower period for debugging, so it seems that ther is some kind of validation ongoing.